tidy3d/components/simulation.py

```diff
@@ -2052,3 +2052,76 @@ def custom_datasets(self) -> List[Dataset]:
         """
         datasets = [src.field_dataset for src in self.sources if isinstance(src, CustomFieldSource)]
         return datasets
+
+    def volumetric_equivalent(self) -> Simulation:
+        """Generate new Simulation wherein any 2D materials are converted to 3D
+        volumetric equivalents."""
+
+        small_num = 1e-5
+
+        # get grid size at monolayer
+        def get_dl(sim, geom, axis):
+            dls = sim.discretize(geom).sizes.to_list[axis]
+            if len(dls) != 1:
+                log.error("2d materials do not support this nonuniform grid.")
```

Comment:
I'm a little confused by this check, so if the dls list along the normal direction of the monolayer has 1 element, it is a uniform grid? but if there is more than one element it is a nonuniform grid? would it be easier to just check if the grid spec along `axis` is uniform? maybe you can explain what the check is looking at here

Reply:
probably this check is unnecessary. would the dls list ever have more than 1 element? the monolayer has 0 thickness

Reply:
Let's just leave it in (in case things change) but use a `Tidy3dException` instead of a `log.error` for consistency. And I think we could change the error message to just be defensive and say something like
```
"More than one grid cell detected across 2D material. Can't generate volumetric equivalent for this simulation. If you received this error, please create an issue in the Tidy3D github repository.
```
to indicate that this is an unexpected bug on our side.